net/src/ipc/transport_wss/tcp.rs

```diff
@@ -0,0 +1,47 @@
+//! abstraction for working with Websocket connections
+//! TcpStream specific functions
+
+use crate::ipc::{
+    transport::{Transport, TransportError, TransportEvent, TransportId, TransportResult},
+    transport_wss::{TransportWss, DEFAULT_HEARTBEAT_WAIT_MS},
+};
+
+impl TransportWss<std::net::TcpStream> {
+    /// convenience constructor for creating a websocket "Transport"
+    /// instance that is based of the rust std TcpStream
+    pub fn with_std_tcp_stream() -> Self {
+        TransportWss::new(|uri| {
+            let socket = std::net::TcpStream::connect(uri)?;
+            socket.set_nonblocking(true)?;
+            Ok(socket)
+        })
+    }
+
+    /// connect and wait for a Connect event response
+    pub fn wait_connect(&mut self, uri: &str) -> TransportResult<TransportId> {
```

Comment:
I don't think anything in here is specific to a TCP stream... should this move to the generic?

Reply:
Totally. Wasn't sure since it was only used for a TcpStream...

Reply:
Moved.